addons/project/wizard/project_share_wizard.py

```diff
@@ -35,14 +35,38 @@ def _compute_resource_ref(self):
             else:
                 wizard.resource_ref = None
 
+    def action_send_mail_post_warning(self):
```

Comment:
This method should be private I think, because if it is public like you did, we can call this method via a rpc call and bypass the process done in `action_send_mail` method.
We could rename this method into `_send_mail` or `_action_send_mail` for instance. 🙂